### DISCUSSION TITLE
What'ss the format about sw_plugin_kafka_producer_config by System Enviroment

### DISCUSSION THREAD

**Discussion by yqjdcyy** [apache/skywalking#11810] [Q&A] [answered]
## Background
- try improve kafka.producer by comppress in lz4
- java.agent in v8.14.0

## Flow
- direct start with agent.args works well
    -Dskywalking.plugin.kafka.producer_config[compression.type]=lz4

- **fail** when try to work on pod in the way of **env**
    - try-1/ sw_plugin_kafka_producer_config {"compression.type":"lz4"}
    - try-2/ sw_plugin_kafka_producer_config "compression.type"="lz4"

## Question
- what's format about `sw_plugin_kafka_producer_config` in the case of system.env
- with pointing the key code maybe be better

**Comment by CodePrometheus** [comment, marked as answer]
I think it may need to be set like this `sw_plugin_kafka_producer_config[compression.type]=lz4` or `SW_PLUGIN_KAFKA_PRODUCER_CONFIG_JSON={"compression.type": "lz4"}`

**Reply by yqjdcyy** [reply]
Thx for your answer

- ENVS `SW_PLUGIN_KAFKA_PRODUCER_CONFIG_JSON={"compression.type": "lz4"}` is good, but **GSON is provided**(not compile) in v8.14.0
- ARGS `-Dskywalking.plugin.kafka.producer_config[compression.type]=lz4` work well , but **NOT WORK for ENVS** with `sw_plugin_kafka_producer_config[compression.type]=lz4`

- with more test.case , when i add the 
 `plugin.kafka.producer_config[compression.type]=${sw_plugin_kafka_producer_config[compression.type]:lz4}`  in /config/agent.config, the agent went to read the ENV even thoung the format is wrong

- maybe you can give me more advise

**Reply by wu-sheng** [reply]
> ENVS SW_PLUGIN_KAFKA_PRODUCER_CONFIG_JSON={"compression.type": "lz4"} is good, but GSON is provided(not compile) in v8.14.0

GSON should have been included in the agent core.

**Comment by wu-sheng** [comment]
`sw_plugin_kafka_producer_config` was a mistake in setting the system env name in lowercase. This is the fix, https://github.com/apache/skywalking-java/pull/668

**Comment by yqjdcyy** [comment]
update config/agent.config/compression.type default to lz4, just like [Dockerfile.provider#L23](https://github.com/apache/skywalking/blob/78434e8b8d769c405ffb96dbcf0b3b22259172f3/test/e2e-v2/cases/kafka/Dockerfile.provider#L23)
`plugin.kafka.producer_config[compression.type]=lz4`
